libraries/rush-lib/src/logic/installManager/WorkspaceInstallManager.ts

```diff
@@ -253,61 +263,44 @@ export class WorkspaceInstallManager extends BaseInstallManager {
         );
         shrinkwrapIsUpToDate = false;
       }
-    }
 
-    // For pnpm pacakge manager, we need to handle dependenciesMeta changes in package.json. See more: https://pnpm.io/package_json#dependenciesmeta
-    // If dependenciesMeta settings is different between package.json and pnpm-lock.yaml, then shrinkwrapIsUpToDate return false.
-    if (this.rushConfiguration.packageManager === 'pnpm') {
-      // First, build a object for dependenciesMeta settings in package.json
-      // key is the package path, value is the dependenciesMeta info for that package
-      const packagePathToDependenciesMetaInPackageJson: { [key: string]: IDependenciesMetaTable } = {};
-      const commonTempFolder: string = this.rushConfiguration.commonTempFolder;
-      const rushJsonFolder: string = this.rushConfiguration.rushJsonFolder;
-
-      // get the relative path from common temp folder to repo root folder
-      const relativeFromTempFolderToRootFolder: string = path.relative(commonTempFolder, rushJsonFolder);
-      for (const rushProject of this.rushConfiguration.projects) {
-        if (!subspace.contains(rushProject)) {
-          // skip processing any project that isn't in this subspace
-          continue;
+      const dependencyMetaList: ReadonlyArray<PackageJsonDependencyMeta> = packageJson.dependencyMetaList;
+      if (dependencyMetaList.length !== 0) {
+        const dependenciesMeta: IDependenciesMetaTable = {};
+        for (const dependencyMeta of dependencyMetaList) {
+          dependenciesMeta[dependencyMeta.name] = {
+            injected: dependencyMeta.injected
+          };
         }
-        const packageJson: PackageJsonEditor = rushProject.packageJsonEditor;
-        const projectRelativeFolder: string = rushProject.projectRelativeFolder;
 
         // get the relative path from common temp folder to package folder, to align with the value in pnpm-lock.yaml
-        const relativePathFromTempFolderToPackageFolder: string =
-          relativeFromTempFolderToRootFolder + '/' + projectRelativeFolder;
-        const dependencyMetaList: ReadonlyArray<PackageJsonDependencyMeta> = packageJson.dependencyMetaList;
-
-        if (dependencyMetaList.length !== 0) {
-          const dependenciesMeta: IDependenciesMetaTable = {};
-          for (const dependencyMeta of dependencyMetaList) {
-            dependenciesMeta[dependencyMeta.name] = {
-              injected: dependencyMeta.injected
-            };
-          }
-          packagePathToDependenciesMetaInPackageJson[relativePathFromTempFolderToPackageFolder] =
-            dependenciesMeta;
-        }
+        const relativePathFromTempFolderToPackageFolder: string = `${relativeFromTempFolderToRootFolder}/${rushProject.projectRelativeFolder}`;
```

Comment:
Did we ever fully enforce that `rushProject.projectRelativeFolder` neither starts nor ends with a slash, and does not contain `\\`?

Reply:
We actually don't. We probably should, but that should be a in a follow-up.